evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -626,3 +626,39 @@ def test_simple_imputer_boolean_nullable_valid_train_empty_test():
     X_t = imp.transform(X_test)
     assert not X_t["a"].isna().any()
     assert isinstance(X_t.ww.logical_types["a"], BooleanNullable)
+
+
+def test_simple_imputer_all_bools_at_fit_with_nans_at_transform():
+    # X_train will be only bool dtypes so the _component_obj won't be fit
+    X_train = pd.DataFrame(
+        {
+            "bools1": pd.Series([True, False, True, True] * 20),
+            "bools2": pd.Series([True, False, True, False] * 20),
+        },
+    )
+    X_train.ww.init(
+        logical_types={
+            "bools1": "Boolean",
+            "bools2": "Boolean",
+        },
+    )
+
+    imp = SimpleImputer()
+    imp.fit(X_train)
+
+    # X_test will be BooleanNullable which will be a problem when _component_obj isn't fit
```

Comment:
I'm confused what this comment is saying here, could you clarify? 

You also explicitly cast the woodwork types to be BooleanNullable, but they're already automatically inferred to be. This isn't a bad thing, it protects us against future changes in woodwork inference, but I'm a bit confused with the context of the comment.

Reply:
This is just the follow-up to the first comment, explaining the test case where we start with `Boolean` types in X_train and then have BooleanNullable types with nans in X_test. Once I switch to convert boolean types instead of calling fit inside of `transform`, this test will stay the same, though maybe I'll add a check that the actual imputed values are the same as the values that would have been imputed if the types were `BooleanNullable` all along